### PR TITLE
Full review of reduce documentation - Issue 169

### PR DESCRIPTION
**Pull request by jdgarciauc3m**
Reviewed all documentation of reduce.
Also simplified the interface.

### CODE REVIEW

File: doc/reduce.md [hunk 1]
```diff
@@ -12,30 +12,104 @@ grppi::reduce(exec, other_arguments...);
 
 There is a single variant of the reduction:
 
-* Sequence reduction: Takes a sequence represented by two iterators and reduces the values.
+* **Sequence reduction without initial value**: Reduces a sequence of values
```
Comment (Drioastorga) [inline doc/reduce.md:15]: I think that this is not true anymore. Now, this interface allways requires the identity value.  This applies also for its description.

File: doc/reduce.md [hunk 2]
```diff
@@ -12,30 +12,104 @@ grppi::reduce(exec, other_arguments...);
...
-The key element of a reduction is the combination **Operation**. The operation may be any C++ callable entity. This operation, is a binary operation taking two values of the same type and combining them into a value of the same type.
+The key element of a reduction is the **Combiner** operation. The **Combiner**
+may be an **HeterogeneousCombiner** or an **HomogeneousCombiner**.
+
+A **Combiner** is any C++ callable entity, that is able to combine two values
+into a single value. The requirements for the **Combiner** are different
+depending on whether the reduction uses an initial value
+(**HeterogeneousCombiner**) or not (**HomogeneousCombiner**).
+
+An **HomogeneousCombiner** `cmb` is any operatingo taking two values `x`and `y`
```
Comment (Drioastorga) [inline doc/reduce.md]: operatingo -> operation

File: doc/reduce.md [hunk 3]
```diff
@@ -12,30 +12,104 @@ grppi::reduce(exec, other_arguments...);
...
+of type `T` and returning a combined value of type `T`, making valid the
+following.
+
+~~~{.cpp}
+T x, y;
+T res = cmb(x,y);
+~~~
+
+An **HeterogeneousCombiner** `cmb` is any operations taking two values `x` and
```
Comment (Drioastorga) [inline doc/reduce.md]: I think that it should be in singular: 
operations -> operation

File: doc/reduce.md [hunk 4]
```diff
@@ -12,30 +12,104 @@ grppi::reduce(exec, other_arguments...);
...
+~~~{.cpp}
+T x;
+U y;
+T res = cmb(x,y);
+~~~
 
 ## Details on reduction variants
 
-### Sequence reduction
+### Sequence reducction without identity
```
Comment (Drioastorga) [inline doc/reduce.md]: This subsection should be removed as this interface is not supported in the current version.

**Review by Drioastorga — CHANGES_REQUESTED** [review]
I have some comments about the reduce documentation that have to be addressed. Additionally, I have an issue for viewing the "collaboration diagram" of the different actual implementations of patterns. I don't know if is a real issue or is my browser but it shows a broken link to an image.

File: doc/reduce.md (hunk 1 above)
Comment (jdgarciauc3m) [reply doc/reduce.md:15]: You are right!

File: doc/reduce.md (hunk 2 above)
Comment (jdgarciauc3m) [reply doc/reduce.md]: OK

File: doc/reduce.md (hunk 3 above)
Comment (jdgarciauc3m) [reply doc/reduce.md]: OK

**Review by Drioastorga — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Solves issue #169

**Pull request merged**